core/api/src/app/wallets/index.ts

```diff
@@ -16,14 +16,35 @@ export * from "./settle-payout-txn"
 export * from "./update-legacy-on-chain-receipt"
 export * from "./update-pending-invoices"
 export * from "./validate"
+export * from "./get-invoice-for-wallet-by-hash"
 
+import { CouldNotFindWalletFromIdError } from "@/domain/errors"
 import { WalletsRepository } from "@/services/mongoose"
 
 export const getWallet = async (walletId: WalletId) => {
   const wallets = WalletsRepository()
   return wallets.findById(walletId)
 }
 
+export const getWalletForAccountById = async ({
+  accountId,
+  walletId,
+}: {
+  accountId: AccountId
+  walletId: WalletId
+}): Promise<Wallet | ApplicationError> => {
+  const wallets = WalletsRepository()
+  const wallet = await wallets.findById(walletId)
+
+  if (wallet instanceof Error) return wallet
+
+  if (wallet.accountId !== accountId) {
```

Comment:
wallet.accountId is probably accountId but should be accountUuid?

I think we may want to do the AccountId/Uuid refactoring (ie: removing the use of `account._id` and solely expose `account.uuid`) before merging this PR?

Reply:
working on this PR: https://github.com/GaloyMoney/galoy/pull/3316